server/service/apple_mdm.go

```diff
@@ -2974,27 +2974,40 @@ func (svc *MDMAppleDDMService) DeclarativeManagement(r *mdm.Request, dm *mdm.Dec
 	}
 	level.Debug(svc.logger).Log("msg", "ddm request received", "endpoint", dm.Endpoint)
 
+	if dm.UDID == "" {
+		return nil, ctxerr.New(r.Context, "missing device id")
+	}
+
+	h, err := svc.ds.HostLiteByIdentifier(r.Context, dm.UDID)
+	if err != nil {
+		return nil, ctxerr.Wrap(r.Context, err, "getting host by identifier")
+	}
+	var tid uint
+	if h.TeamID != nil {
+		tid = *h.TeamID
+	}
+
 	switch {
 	case dm.Endpoint == "tokens":
+		level.Debug(svc.logger).Log("msg", "received tokens request")
+		// TODO: Should we record the checkin for all endpoints or just tokens?
 		if err := svc.ds.MDMAppleRecordDeclarativeCheckIn(r.Context, dm.UDID, dm.Raw); err != nil {
 			return nil, ctxerr.Wrap(r.Context, err, "recording declarative checkin")
 		}
-		// TODO(sarah): handle tokens
-		level.Debug(svc.logger).Log("msg", "received tokens request")
-		return nil, nil
+
+		return svc.handleTokens(r.Context, tid)
 
 	case dm.Endpoint == "declaration-items":
-		// TODO(sarah): handle declaration-items
 		level.Debug(svc.logger).Log("msg", "received declaration-items request")
-		return nil, nil
+		return svc.handleDeclarationItems(r.Context, tid)
```

Comment:
something I just realized while reviewing this: since declarations can have labels, you'll need to pass the host (or the host.id or whatever) and adjust the query.

unfortunately, we need to do the same for `tokens` (and probably discard the view?) because a host might get a new label applied to it.